src/averageAges.js

```diff
@@ -14,12 +14,26 @@
  *
  * @return {number}
  */
+const MALE = 'm';
+const FEMALE = 'f';
+
 function calculateMenAverageAge(people, century) {
-  // write code here
-  // learn how to use array methods like .filter .map .some .every .find .reduce
-  // avoid using loop and forEach
-  // replace `if ()` statement with &&, || or ?:
-  // without nesting
+  const manOnly = people.filter(person => person.sex === MALE);
+  let sumManAge;
+
+  if (!century) {
+    sumManAge = manOnly.reduce((sumAge, person) =>
+      sumAge + (person.died - person.born), 0);
+
+    return +(sumManAge / manOnly.length).toFixed(2);
+  }
+
+  const menInCentury = manOnly.filter(man =>
+    Math.ceil(man.died / 100) === century);
```

Comment:
You can filter by century conditionally in the first filter and then you can get rid of `if ()` statement